qiita_db/base.py

```diff
@@ -166,13 +166,26 @@ def __init__(self, id_):
 
         Parameters
         ----------
-        id_: the object identifier
+        id_: int, long, or str
+            the object identifier
 
         Raises
         ------
         QiitaDBUnknownIDError
             If `id_` does not correspond to any object
         """
+        id_type = type(id_)
+        if not isinstance(id_, (int, long, str, unicode)):
+            raise TypeError("id_ must be a numerical or text type (not %s) "
+                            "when instantiating %s" % (str(id_type),
+                                                       str(self.__class__)))
+
+        if id_type in (str, unicode):
+            if id_.isdigit():
```

Comment:
This is a nice compromise for User object. Could you put in a comment about why it's needed?